### PR TITLE
Fix metrics

### LINKED ISSUE #353: Tests generate lots of warnings
Specifically, `apply_transform` and 'metrics` tests. Fix the code to remove warnings.

### PR DESCRIPTION
**Pull request by cdtn**
Main changes:

1. Multiclass averaging for binary case corrected.
2. Division by zero and other warnings fixed.

This closes #353.

### CODE REVIEW

File: batchflow/tests/pytest.ini [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[pytest]
+markers =
+    slow: marks tests as slow (deselect with '-m "not slow"')
+    serial
```
Comment (Dimonovez) [inline batchflow/tests/pytest.ini]: Add new line

File: batchflow/models/metrics/classify.py [hunk 2]
```diff
@@ -359,7 +339,8 @@ def diagnostics_odds_ratio(self, *args, **kwargs):
                               when_zero=(np.inf, 0))
 
     def f1_score(self, *args, **kwargs):
-        return 2 / (1 / self.recall(*args, **kwargs) + 1 / self.precision(*args, **kwargs))
+        pair = self.recall(*args, **kwargs), self.precision(*args, **kwargs)
```
Comment (SergeyTsimfer) [inline batchflow/models/metrics/classify.py]:
```suggestion
        recall, precision = self.recall(*args, **kwargs), self.precision(*args, **kwargs)
```

File: batchflow/tests/metrics_test.py [hunk 3]
```diff
@@ -42,7 +42,7 @@
 PROBA = np.eye(NUM_CLASSES)[LABELS]
 
 # Logit function gives ±infs on degenerate case of 0s and 1s but it's works fine for sigmoid function.
```
Comment (SergeyTsimfer) [inline batchflow/tests/metrics_test.py]:
```suggestion
# Logit function gives ±infs on degenerate case of 0s and 1s but works fine for sigmoid function.
```

File: batchflow/models/metrics/utils.py [hunk 4]
```diff
@@ -40,3 +44,20 @@ def get_components(inputs, batch=True):
...
+    """ Compute the arithmetic mean along 0 axis ignoring infs, when there is
+    at least one finite number along averaging axis. Done via np.nanmean()
+    while temporarily replacing np.inf with np.nan.
+    """
+    if isinstance(arr, list):
+        arr = np.array(arr)
+    arr[np.isinf(arr)] = np.nan
+    # Mean of empty slice is expected to be np.nan, so the warning is redundant
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=RuntimeWarning)
```
Comment (SergeyTsimfer) [inline batchflow/models/metrics/utils.py]: It would be better to ignore specific message

File: batchflow/models/metrics/utils.py [hunk 5]
```diff
@@ -40,3 +44,20 @@ def get_components(inputs, batch=True):
...
+        arr = np.nanmean(arr, axis=axis)
+    if np.isscalar(arr):
+        return np.inf if np.isnan(arr) else arr
+    arr[np.isnan(arr)] = np.inf
```
Comment (Dimonovez) [inline batchflow/models/metrics/utils.py]: Use numpy masked array, specifically `numpy.ma.masked_invalid`

**Pull request merged**